test/CrateTest/PDO/Http/ServerPoolTest.php

```diff
@@ -0,0 +1,238 @@
+                ],
+            ],
+            [
+                [
+                    PDO::CRATE_ATTR_SSL_MODE => PDO::CRATE_ATTR_SSL_MODE_REQUIRED,
+                    PDO::CRATE_ATTR_SSL_KEY  => 'foo.pem',
+                ],
+                [
+                    'base_uri'              => 'https://localhost:4200',
+                    RequestOptions::SSL_KEY => 'foo.pem',
+                ],
+            ],
+            [
+                [
+                    PDO::CRATE_ATTR_SSL_MODE         => PDO::CRATE_ATTR_SSL_MODE_REQUIRED,
+                    PDO::CRATE_ATTR_SSL_KEY          => 'foo.pem',
+                    PDO::CRATE_ATTR_SSL_KEY_PASSWORD => 'foo',
+                ],
+                [
+                    'base_uri'              => 'https://localhost:4200',
+                    RequestOptions::SSL_KEY => ['foo.pem', 'foo'],
+                ],
+            ],
+            [
+                [
+                    PDO::CRATE_ATTR_SSL_MODE => PDO::CRATE_ATTR_SSL_MODE_REQUIRED,
+                    PDO::CRATE_ATTR_SSL_CERT => 'foo.pem',
+                ],
+                [
+                    'base_uri'           => 'https://localhost:4200',
+                    RequestOptions::CERT => 'foo.pem',
+                ],
+            ],
+            [
+                [
+                    PDO::CRATE_ATTR_SSL_MODE          => PDO::CRATE_ATTR_SSL_MODE_REQUIRED,
+                    PDO::CRATE_ATTR_SSL_CERT          => 'foo.pem',
+                    PDO::CRATE_ATTR_SSL_CERT_PASSWORD => 'foo',
+                ],
+                [
+                    'base_uri'           => 'https://localhost:4200',
+                    RequestOptions::CERT => ['foo.pem', 'foo'],
+                ],
+            ],
+        ];
+    }
+
+    /**
+     * @dataProvider pdoOptions
+     *
+     * @param array $options
+     * @param array $expected
+     */
+    public function testGuzzleClientOptionTest(array $options, array $expected)
+    {
+        foreach ($options as $attr => $val) {
+            $this->pdo->setAttribute($attr, $val);
+        }
+
+        $expectedWithDefaults = array_merge([
+            RequestOptions::TIMEOUT         => 0.0,
+            RequestOptions::CONNECT_TIMEOUT => 0.0,
+            RequestOptions::JSON            => [
+                'stmt' => 'query',
+                'args' => [],
+            ],
+
+            RequestOptions::HEADERS => [
+                'Default-Schema' => 'doc',
+            ],
+
+            RequestOptions::AUTH => null,
+            'base_uri'           => 'http://localhost:4200',
+        ], $expected);
+
+        $body = json_encode(['rows' => [], 'cols' => [], 'duration' => 0, 'rowcount' => 0]);
+
+        $this->client
+            ->expects($this->once())
+            ->method('request')
+            ->with('POST', '/_sql', $expectedWithDefaults)
+            ->willReturn(new Response(200, [], $body));
+
+        $this->serverPool->execute('query', []);
```

Comment:
No real assertion here besides: `->expects($this->once())` and `->with('POST', '/_sql', $expectedWithDefaults)`. The `execute()` interface returns a `CollectionInterface`: should probably fill it with something (anything) and verify the return value here for equality.